### PR TITLE
enable backend DB options in cronos

### LINKED ISSUE #706: Problem: cronos mainnet node is unable to export genesis
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by JayT106**
Closes #706

Adding a new flag `maxopenfiles` for setting up the backend DB `max open files`. 
The current rocksdb (in tm-db v0.6.7) is using `4096` as the default value, it affects the backend DB in the `application` and the `snapshot` folder 

Note: for saving the memory usage, we might set this flag to `1000` (the default value in tm-db v0.6.6)
i.e.
```
./cronosd start --maxopenfiles 1000
./cronosd export --maxopenfiles 1000
```

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Comment by JayT106** [comment]
missing nix mod update

File: go.mod [hunk 1]
```diff
@@ -156,7 +158,9 @@ require (
 replace (
 	// TODO: fix keyring upstream
 	github.com/99designs/keyring => github.com/crypto-org-chain/keyring v1.1.6-fixes
-	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.45.8
+
+	// TODO: might use crypto-org repo as the refrence path
+	github.com/cosmos/cosmos-sdk => github.com/jayt106/cosmos-sdk v0.45.8-db-opts
```
Comment (yihuang) [inline go.mod:163]: Should create the PR to upstream?

Comment (JayT106) [reply go.mod:163]: I will create it, but not sure the SDK team will want to change it because the `tm-db` version changed.

Comment (yihuang) [reply go.mod:163]: Yeah, usually we create the PR to main first, then try to backport

**Comment by JayT106** [comment]
created PR in the SDK `main`
https://github.com/cosmos/cosmos-sdk/pull/13406

**Comment by yihuang** [comment]
I think the best way to going forward this is control rocksdb options in different commands, even customizable with cli flags or configs, which is possible now, with recent `NewRocksdbWithRawDB` api in our forked tm-db.
So I'm closing this, @JayT106

**Pull request closed without merging**